src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_release.kt

```diff
@@ -0,0 +1,509 @@
+package uk.gov.justice.hmpps.prison.api.resource.impl
+
+import io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.MediaType
+import org.springframework.test.web.reactive.server.StatusAssertions
+import uk.gov.justice.hmpps.prison.api.model.InmateDetail
+import uk.gov.justice.hmpps.prison.service.DataLoaderTransaction
+import uk.gov.justice.hmpps.prison.util.builders.ExternalServiceBuilder
+import uk.gov.justice.hmpps.prison.util.builders.OffenderBookingBuilder
+import uk.gov.justice.hmpps.prison.util.builders.OffenderBuilder
+import uk.gov.justice.hmpps.prison.util.builders.OffenderNoPayPeriodBuilder
+import uk.gov.justice.hmpps.prison.util.builders.OffenderPayStatusBuilder
+import uk.gov.justice.hmpps.prison.util.builders.OffenderProgramProfileBuilder
+import uk.gov.justice.hmpps.prison.util.builders.ServiceAgencySwitchBuilder
+import uk.gov.justice.hmpps.prison.util.builders.getBedAssignments
+import uk.gov.justice.hmpps.prison.util.builders.getCaseNotes
+import uk.gov.justice.hmpps.prison.util.builders.getKeyDateAdjustments
+import uk.gov.justice.hmpps.prison.util.builders.getMovements
+import uk.gov.justice.hmpps.prison.util.builders.getOffenderNoPayPeriods
+import uk.gov.justice.hmpps.prison.util.builders.getOffenderPayStatus
+import uk.gov.justice.hmpps.prison.util.builders.getOffenderProgramProfiles
+import uk.gov.justice.hmpps.prison.util.builders.getSentenceAdjustments
+import uk.gov.justice.hmpps.prison.util.builders.release
+import uk.gov.justice.hmpps.prison.util.builders.transferOutToCourt
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+class OffenderResourceIntTest_release : ResourceTest() {
+
+  @Autowired
+  private lateinit var dataLoaderTransaction: DataLoaderTransaction
+
+  @Nested
+  @DisplayName("POST /offenders/{offenderNo}/release")
+  inner class ReleaseOffender {
```

Comment:
The main change is a new set of integration tests for releasing an offender. This tests all aspects of the existing endpoint as a precursor to refactoring the service.